ada_feeding/ada_feeding/behaviors/blackboard_behavior.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Generic behavior with additional functionality for blackboard
+remappings (inspired by BehaviorTree.CPP)
+"""
+
+# Standard imports
+from typing import Any
+
+# Third-party imports
+import py_trees
+
+# Local imports
+from ada_feeding.helpers import BlackboardKey
+
+
+class BlackboardBehavior(py_trees.behaviour.Behaviour):
+    """
+    TODO
+    """
+
+    def __init__(self, name: str, ns: str = "/") -> None:
+        """
+        Initialize the behavior and blackboard
+
+        Parameters
+        ----------
+        name: Behavior name
+        ns: Blackboard namespace (usually the name of the tree / subtree)
+        """
+        super().__init__(name=name)
+        self.blackboard = super().attach_blackboard_client(name=name, namespace=ns)
+        self.remap = {}
+        self.inputs = {}
+        self.outputs = {}
+
+    def blackboard_inputs(self, **kwargs) -> None:
+        """
+        Define and register all blackboard input keys.
+
+        Generally this is done in the subclass as follows:
+        <key>: Union[BlackboardKey, <type>] = <default>
+
+        Call from the subclass using:
+        super().blackboard_inputs(**{key: value for key,
+            value in locals().items() if key != 'self'})
+
+        Constants will be stored in the local dict.
+        """
+        for key, value in kwargs.items():
+            if isinstance(value, BlackboardKey):
+                self.blackboard.register_key(
+                    key=value, access=py_trees.common.Access.READ
+                )
+                self.remap[key] = value
+            else:
+                self.inputs[key] = value
+
+    def blackboard_outputs(self, **kwargs) -> None:
+        """
+        Define and register all blackboard output keys.
+
+        Generally this is done in the subclass as follows:
+        <key>: BlackboardKey = <default>
+
+        Call from the subclass using:
+        super().blackboard_inputs(**{key: value for key,
+            value in locals().items() if key != 'self'})
+        """
+        for key, value in kwargs.items():
+            if isinstance(value, BlackboardKey):  # Catches None
+                self.blackboard.register_key(
+                    key=value, access=py_trees.common.Access.WRITE
+                )
+            self.outputs[key] = value
+
+    def blackboard_exists(self, key: str) -> bool:
+        """
+        Check if a key is set in the blackboard or available locally.
```

Comment:
Add to `blackboard_exists`, `blackboard_get`, `blackboard_write` docstrings: `Raises KeyError if the key has not been defined in blackboard_inputs/outputs`